### PR TITLE
Properly get coverage from all tests.

### PR DESCRIPTION
**Pull request by yossigo**
* Use --coverage when building CFFI module.
* Use out-of-line CFFI config to avoid re-building the module, causing
  libgcov to complain about differing timestamps across different runs.
* Makefile adjustments to built CFFI as needed.
* Remove raft_get_snapshot_entry_idx, a prototype for an undefined
  function that breaks CFFI.

### CODE REVIEW

**Comment by yossigo** [comment]
@tezc Following our discussion today, this addresses coverage from all tests (test_fuzzer and test_virtraft). Also cleans up the CFFI stuff a bit, but I guess it may require some adjustments on the CMake side.

File: tests/raft_cffi_builder.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+def load(fname):
+    return '\n'.join(
+        [line for line in subprocess.check_output(
+            ["gcc", "-E", fname]).decode('utf-8').split('\n')])
+
+if __name__ == '__main__':
+
+    ffibuilder = cffi.FFI()
+    ffibuilder.set_source(
+        "tests.raft_cffi",
```
Comment (tezc) [inline tests/raft_cffi_builder.py:13]: Prefix was "ffi_tests" here to clean up on `clean` target via `rm -f ffi_tests.*`. It doesn't cleanup raft_cffi.c and raft_cffi_bla_bla.so under `tests/` right now. Maybe we keep following similar approach here?

Comment (yossigo) [reply tests/raft_cffi_builder.py:13]: Thanks, fixed that!

**Comment by tezc** [comment]
@yossigo I couldn't adapt cmake to these changes on the first try, I'll create a follow up PR for it later.

**Review by tezc — APPROVED** [review]

**Pull request merged**